code/modules/mapping/mapping_helpers.dm

```diff
@@ -236,31 +236,79 @@ INITIALIZE_IMMEDIATE(/obj/effect/mapping_helpers/no_lava)
 	name = "Dead Body placer"
 	late = TRUE
 	icon_state = "deadbodyplacer"
-	var/bodycount = 2 //number of bodies to spawn
+	/// number of bodies to spawn
+	var/bodycount = 1
+	/// -1: area search (VERY expensive - do not use this in maint/ruin type)
+	/// 0: spawns onto itself
+	/// +1: turfs from this dead body placer
+	var/search_view_range = 0
+	/// the list of container typepath which accepts dead bodies
+	var/list/accepted_list = list(
+		/obj/structure/bodycontainer/morgue,
+		/obj/structure/closet
+	)
+
+/// as long as this body placer is contained within medbay morgue, this is fine to be expensive.
+/// DO NOT USE this outside of medbay morgue
+/obj/effect/mapping_helpers/dead_body_placer/medbay_morgue
+	bodycount = 2
+	accepted_list = list(/obj/structure/bodycontainer/morgue)
+	search_view_range = -1
+
+/obj/effect/mapping_helpers/dead_body_placer/ruin_morgue
+	bodycount = 2
+	accepted_list = list(/obj/structure/bodycontainer/morgue)
+	search_view_range = 7
+
+/obj/effect/mapping_helpers/dead_body_placer/maint_fridge
+	bodycount = 2
+	accepted_list = list(/obj/structure/closet)
+	search_view_range = 0
 
 /obj/effect/mapping_helpers/dead_body_placer/LateInitialize()
-	var/area/area = get_area(src)
-	var/list/trays = list()
-	for (var/thingy in area.contents)
-		if (istype(thingy, /obj/structure/bodycontainer/morgue))
-			trays += thingy
-	if(!length(trays))
-		log_mapping("[src] at [x],[y] could not find any morgues.")
-		return
-	for (var/i = 1 to bodycount)
-		create_corpse(pick(trays))
+	var/area/current_area = get_area(src)
+	var/list/found_container = list()
+
+	// search_view_range
+	//   [Negative]: area search, get_contained_turfs()
+	if(search_view_range < 0)
+		for(var/turf/each_turf in current_area.get_contained_turfs())
+			for(var/obj/each_container in each_turf)
+				if(get_area(each_container) != current_area)
+					continue // we don't want to put a deadbody to a wrong area
```

Comment:
This isn't possible

Reply:
oh, yeah, I did some braindead copypasta here